Add Java code style guide by Google.

guidomaiolams opened the pull request:
To check it visually follow this link https://github.com/guidomaiolams/code-style-guides/tree/master/Java

AlphaGit commented on Java/README.md:
```diff
@@ -0,0 +1,687 @@
+      lastThing();
+    }
+  }
+};
```
Minor: consider indenting these or using ``` so that it is properly formatted as code. (This applies to all code samples in this file.)

guidomaiolams replied:
Fixed ! wasnt sure if this syntax works for java, but seems yes .. thanks!

AlphaGit commented:
@guidomaiolams Minor: consider adapting the PR title and description to English -- we do that so we keep ourselves available to the community if they want to participate.

pbarrios commented:
Wow, this awesome. Thanks a lot for it. Will review it later but for the moment just a minor comment: Can we please switch the PR description to English? (just to keep consistency with the rest of the repo documentation)

AlphaGit commented:
Thank you for the contribution! I left a suggestion but it overall looks really good!

matiasbeckerle commented:
Thanks for the contribution -- should we add some clause alluding that this was extracted from Google?

guidomaiolams commented:
> Thanks for the contribution -- should we add some clause alluding that this was extracted from Google?

Done ! I added a Note at the top, i just extracted from Google's style guide, did the migration to markup syntax and added the checkstyle link... very useful in order to do all these auto checks in the IDE. Thanks !
